File: src/android/app/src/main/java/org/yuzu/yuzu_emu/utils/DirectoryStateReceiver.java
package org.yuzu.yuzu_emu.utils;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import org.yuzu.yuzu_emu.utils.DirectoryInitialization.DirectoryInitializationState;

public class DirectoryStateReceiver extends BroadcastReceiver {
    Action1<DirectoryInitializationState> callback;

    public DirectoryStateReceiver(Action1<DirectoryInitializationState> callback) {
        this.callback = callback;
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        DirectoryInitializationState state = (DirectoryInitializationState) intent
                .getSerializableExtra(DirectoryInitialization.EXTRA_STATE);
        callback.call(state);
    }
}

File: src/android/app/src/main/java/org/yuzu/yuzu_emu/utils/DirectoryStateReceiver.kt
// SPDX-FileCopyrightText: 2023 yuzu Emulator Project
// SPDX-License-Identifier: GPL-2.0-or-later

package org.yuzu.yuzu_emu.utils

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import org.yuzu.yuzu_emu.utils.DirectoryInitialization.DirectoryInitializationState

class DirectoryStateReceiver(var callback: (DirectoryInitializationState) -> Unit) :
    BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val state = intent
            .getSerializableExtra(DirectoryInitialization.EXTRA_STATE) as DirectoryInitializationState
        callback.invoke(state)
    }
}
